### PR TITLE
Populate `face_face_connectivity`

### LINKED ISSUE #828: Populate `face_face_connectivity`
### Proposed new feature or change:

Feature to populate Face-Face connectivity based on what faces share an edge.

### PR DESCRIPTION
**Pull request by aaronzedwick**
Closes #828 

## Overview

Adds the ability for the `face_face_connectivity` attribute to be constructed and populated in UXarray if it doesn't exist.
## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Testing**
- [x] Adequate tests are created if there is new functionality
- [x] Tests cover all possible logical paths in your function
- [x] Tests are not too basic (such as simply calling a function and nothing else)

### CODE REVIEW

**Review by philipc2 — COMMENTED** [review]
Can we address #845 with this PR too?

**Review by philipc2 — COMMENTED** [review]
Would you be interested in creating an ASV benchmark suite for connectivity construction, which would test the execution time of the construction methods?

For reference, this is what a benchmark file looks like. It's very similar to how our unit tests are written. https://github.com/UXARRAY/uxarray/blob/main/benchmarks/face_bounds.py

Design wise, something like the following could work
- File: `construct_connectivity.py`
- Class: `FaceFaceConnectivity`
- Benchmark Methods: `time_construct`

Let me know if you are interested!

**Comment by rajeeja** [comment]
face face connectivity means adjacent faces, correct? make sure you check for corner cases, when there is only once face etc.

**Review by rajeeja — APPROVED** [review]
looks good, possible to vectorize the loops?
as @philipc2 timining this would be a good idea.

**Comment by aaronzedwick** [comment]
> Would you be interested in creating an ASV benchmark suite for connectivity construction, which would test the execution time of the construction methods?
> 
> For reference, this is what a benchmark file looks like. It's very similar to how our unit tests are written. https://github.com/UXARRAY/uxarray/blob/main/benchmarks/face_bounds.py
> 
> Design wise, something like the following could work
> 
> * File: `construct_connectivity.py`
> * Class: `FaceFaceConnectivity`
> * Benchmark Methods: `time_construct`
> 
> Let me know if you are interested!

Yeah I am definitely interested, I would love to learn how to make the benchmarking tests!

**Comment by philipc2** [comment]
> > Would you be interested in creating an ASV benchmark suite for connectivity construction, which would test the execution time of the construction methods?
> > For reference, this is what a benchmark file looks like. It's very similar to how our unit tests are written. https://github.com/UXARRAY/uxarray/blob/main/benchmarks/face_bounds.py
> > Design wise, something like the following could work
> > 
> > * File: `construct_connectivity.py`
> > * Class: `FaceFaceConnectivity`
> > * Benchmark Methods: `time_construct`
> > 
> > Let me know if you are interested!
> 
> Yeah I am definitely interested, I would love to learn how to make the benchmarking tests!

You should be able to use the majority of the code from the file I provided as a skeleton. I would start with timing benchmarks for now, and we can look into other benchmarks as well.

https://asv.readthedocs.io/en/stable/writing_benchmarks.html

File: uxarray/grid/connectivity.py [hunk 1]
```diff
@@ -433,3 +433,42 @@ def get_face_node_partitions(n_nodes_per_face):
...
+    # Convert to an array and pad it with fill values
+    face_face_conn = list(face_neighbors.values())
+    face_face_connectivity = [
+        np.pad(
+            arr, (0, grid.n_max_face_edges - len(arr)), constant_values=INT_FILL_VALUE
+        )
+        for arr in face_face_conn
+    ]
+
+    return face_face_connectivity
```
Comment (philipc2) [inline uxarray/grid/connectivity.py:474]: Once @aaronzedwick has written a timing benchmark for this, we can then decide if we need to look into further optimization.

**Review by philipc2 — APPROVED** [review]

**Pull request merged**